review/js/script.js

```diff
@@ -1,167 +1,167 @@
+    } else if (index === slider.children.length - 1) {
+        nextButton.classList.add("carousel__button--disabled");
+        prevButton.classList.remove("carousel__button--disabled");
+    } else {
+        nextButton.classList.remove("carousel__button--disabled");
+        prevButton.classList.remove("carousel__button--disabled");
+    }
+};
+
+// funzione che si occupa della gestione del click sui bottoni
+const handleClick = function (event, dir) {
+    if (isOpen) {
+        const width = slider.children[0].getBoundingClientRect().width;
+        indexs.forEach((node) => node.classList.remove("carousel__index--active"));
+        cancelAnimationFrame(timing);
+
+        if (dir === -1) {
+            index === 0 ? (index = 0) : index--;
+            manageButtons();
+            indexs[index].classList.add("carousel__index--active");
+            return (() => animate(dir, width, index))();
+        } else {
+            index === slider.children.length - 1 ? (index = slider.children.length - 1) : index++;
+            manageButtons();
+            indexs[index].classList.add("carousel__index--active");
+            return (() => animate(dir, width, index))();
+        }
+    }
+};
+
+manageButtons();
+prevButton.addEventListener("click", (event) => handleClick(event, (dir = -1)));
+nextButton.addEventListener("click", (event) => handleClick(event, (dir = 1)));
+
+// gestione dei piccoli dots accanto ai bottoni (gli ho chiamati indici perchè di base sono quelli 😁)
+indexs.forEach((node) =>
+    node.addEventListener("click", (event) => {
+        cancelAnimationFrame(timing);
+        indexs.forEach((node) => node.classList.remove("carousel__index--active"));
+        const idx = [...indexs].indexOf(node);
+        const dir = idx < index ? -1 : 1;
+        const width = slider.children[0].getBoundingClientRect().width;
+        index = idx;
+
+        manageButtons();
+        animate(dir, width, index);
+        indexs[index].classList.add("carousel__index--active");
+    })
+);
+
+// funzione che si occupa della navbar (quando apparire o scomparire)
+main.addEventListener("wheel", (event) => {
+    if (event.deltaY < 0 && !isOpen) header.classList.remove("header--hide");
+    else if (!isOpen) header.classList.add("header--hide");
+});
+
+// gestione e animazione del bottone che apre il menu
+toggle.addEventListener("click", (event) => {
+    isOpen = !isOpen;
+
+    if (isOpen) {
+        toggle.classList.add("toggle--open");
+        menu.classList.add("menu--visible");
+        header.firstElementChild.classList.add("header__background--transparent");
+    } else {
+        toggle.classList.remove("toggle--open");
+        menu.classList.add("menu--invisible");
+        setTimeout(() => {
+            menu.classList.remove("menu--visible");
+            menu.classList.remove("menu--invisible");
+            header.firstElementChild.classList.remove("header__background--transparent");
+        }, 500);
```

Comment:
Questo timeout andrebbe cancellato nel caso in cui l'utente clicchi di nuovo sul pulsante, altrimenti il menu si chiude ma è in realtà aperto e si ha un bug visivo